README.md

```diff
@@ -183,7 +183,7 @@ float minValue = vget_lane_f32(minOfMinOfHalfs, 0);
 - ternary operator: use vector comparison (for example **vcltq_f32** for *less than* comparison)
 ```c
 float32x4_t v1 = { 1.0, 0.0, 1.0, 0.0 }, v2 = { 0.0, 1.0, 1.0, 0.0 };
-float32x4_t mask = vcltq_f32(v1, v2);  // v1 < v2
+uint32x4_t mask = vcltq_f32(v1, v2);  // v1 < v2
```

Comment:
I'm not sure I understand this. `vcltq_f32` deals with floats?

Reply:
By searching the [official online documentation](https://developer.arm.com/architectures/instruction-sets/intrinsics/#f:@navigationhierarchiessimdisa=[Neon]&q=vcltq_f32), we can see the return type is `uint32x4_t` instead of `float32x4_t`:

![图片](https://user-images.githubusercontent.com/3831847/159103665-9a968227-a831-43be-ad01-a3943851e12d.png)

We may also dump the returned value, as `float32x4_t` and as `uint32x4_t`. Actually they should be same in binary format, but for printing, if we see `-nan` for a float value, it is not straightforward, but if we see `4294967295` for a unsigned int, then it is much easier, it 
```c++
void test_vcltq()
{
#if __ARM_NEON
    float32x4_t v1 = { 1.0, 0.0, 1.0, 0.0 };
    float32x4_t v2 = { 0.0, 1.0, 1.0, 0.0 };

    {
        float32x4_t mask = vcltq_f32(v1, v2);  // v1 < v2
        float mask_data[4];
        vst1q_f32(mask_data, mask);
        fprintf(stderr, "with float32x4_t as return type, got:");
        for (int i = 0; i < 4; i++) {
            fprintf(stderr, "%f, ", mask_data[i]);
        }
        fprintf(stderr, "\n");
    }

    {
        uint32x4_t mask = vcltq_f32(v1, v2);  // v1 < v2
        uint32_t mask_data[4];
        vst1q_u32(mask_data, mask);
        fprintf(stderr, "with uint32x4_t as return type, got:");
        for (int i = 0; i < 4; i++) {
            fprintf(stderr, "%u, ", mask_data[i]);
        }
        fprintf(stderr, "\n");
    }

    fprintf(stderr, "%u", 0xffffffff);
#endif
}
```

>with float32x4_t as return type, got:0.000000, -nan, 0.000000, 0.000000, 
>with uint32x4_t as return type, got:0, 4294967295, 0, 0, 
>4294967295

Reply:
Thanks for the explanation, makes sense.